hudi-spark-datasource/hudi-spark3/src/main/scala/org/apache/spark/sql/hudi/catalog/HoodieCatalog.scala

```diff
@@ -206,29 +236,17 @@ class HoodieCatalog extends DelegatingCatalogExtension
     val newPartitionColumns = partitionColumns
     val newBucketSpec = maybeBucketSpec
 
-    val isByPath = isPathIdentifier(ident)
-
-    val location = if (isByPath) Option(ident.name()) else Option(allTableProperties.get("location"))
-    val id = ident.asTableIdentifier
-
-    val locUriOpt = location.map(CatalogUtils.stringToURI)
-    val existingTableOpt = getExistingTableIfExists(id)
-    val loc = locUriOpt
-      .orElse(existingTableOpt.flatMap(_.storage.locationUri))
-      .getOrElse(spark.sessionState.catalog.defaultTablePath(id))
     val storage = DataSource.buildStorageFormatFromOptions(writeOptions.--(needFilterProps))
-      .copy(locationUri = Option(loc))
-    val tableType =
-      if (location.isDefined) CatalogTableType.EXTERNAL else CatalogTableType.MANAGED
```

Comment:
can we call deduceTableLocationURIAndTableType in `createHoodieTable ` and HoodieStageTable instead of call it before `createHoodieTable` and HoodieStageTable?

Reply:
You mean -- HoodieStageTable invokes `deduceTableLocationURIAndTableType` to get the "path" and then invokes `createHoodieTable`. While inside `createHoodieTable` invokes `deduceTableLocationURIAndTableType` AGAIN to get the "path";

I'd prefer to invoke `deduceTableLocationURIAndTableType` only once between HoodieCatalog and HoodieStageTable. The deducing logic doesn't make much performance cost. But I'd rather keep the invoking logic straightforward.

Reply:
For the first version of this PR, I construct a whole `HoodieCatalogTable` first, and then pass it to` HoodieStageTable` and `createHoodieTable` -- but I think it might be better to only pass the necessary information(location and table type) to `HoodieStageTable`.